src/add-text-track-data.js

```diff
@@ -102,17 +102,38 @@ const addTextTrackData = function(sourceHandler, captionArray, metadataArray) {
       let cues = sourceHandler.metadataTrack_.cues;
       let cuesArray = [];
 
-      for (let j = 0; j < cues.length; j++) {
-        cuesArray.push(cues[j]);
-      }
-      cuesArray.sort((first, second) => first.startTime - second.startTime);
-
-      for (let i = 0; i < cuesArray.length - 1; i++) {
-        if (cuesArray[i].endTime !== cuesArray[i + 1].startTime) {
-          cuesArray[i].endTime = cuesArray[i + 1].startTime;
+      // Create a copy of the TextTrackCueList...
+      // ...disregarding cues with a falsey value
+      for (let i = 0; i < cues.length; i++) {
+        if (cues[i]) {
+          cuesArray.push(cues[i]);
         }
       }
-      cuesArray[cuesArray.length - 1].endTime = videoDuration;
+
+      // Group cues by their startTime value
+      let cuesGroupedByStartTime = cuesArray.reduce((obj, cue) => {
```

Comment:
We could avoid having to need this `reduce` call completely by creating the `cuesGroupedByStartTime` object above in the for loop instead of populating the `cuesArray` that is only used for this reduce call.